spec/lib/request_pool_spec.rb

```diff
@@ -48,16 +48,24 @@
       expect(subject.size).to be > 1
     end
 
-    it 'closes idle connections' do
-      stub_request(:get, 'http://example.com/').to_return(status: 200, body: 'Hello!')
+    context 'with an idle connection' do
+      before do
+        stub_const('RequestPool::MAX_IDLE_TIME', 0.2) # Lower the idle time limit to 0.2 seconds
+        stub_const('RequestPool::REAPER_FREQUENCY', 0.1) # Run the reaper every 0.1 seconds
+        stub_request(:get, 'http://example.com/').to_return(status: 200, body: 'Hello!')
+      end
 
-      subject.with('http://example.com') do |http_client|
-        http_client.get('/').flush
+      it 'closes the connections' do
+        subject.with('http://example.com') do |http_client|
+          http_client.get('/').flush
+        end
+
+        expect { wait_for_idle_timeout }.to change(subject, :size).from(1).to(0)
       end
 
-      expect(subject.size).to eq 1
-      sleep RequestPool::MAX_IDLE_TIME + 30 + 1
-      expect(subject.size).to eq 0
+      def wait_for_idle_timeout
+        sleep RequestPool::MAX_IDLE_TIME + 0.1 # Wait slightly longer than idle time
+      end
```

Comment:
Pushed change which bumps max idle timeout to 1s, and then also adjusts the sleep to be based off the idle time plus a multiple of the reaper timeout, so that we are more certain to get a reaper cycle after the idle time but before our sleep ends.